libs/vkd3d/command.c

```diff
@@ -1331,12 +1331,32 @@ static ULONG STDMETHODCALLTYPE d3d12_shared_fence_Release(d3d12_fence_iface *ifa
 {
     struct d3d12_shared_fence *fence = shared_impl_from_ID3D12Fence1(iface);
     ULONG refcount = InterlockedDecrement(&fence->refcount);
+    struct vkd3d_shared_fence_waiting_event *current, *e;
+    struct d3d12_device *device = fence->device;
+    BOOL is_running;
 
     TRACE("%p decreasing refcount to %u.\n", fence, refcount);
 
     if (!refcount)
     {
-        struct d3d12_device *device = fence->device;
+        pthread_mutex_lock(&fence->mutex);
+        is_running = fence->is_running;
+        fence->is_running = false;
+        pthread_mutex_unlock(&fence->mutex);
+        if (is_running)
+        {
+            pthread_cond_signal(&fence->cond_var);
```

Comment:
While pthread_cond_signal outside the lock here is technically OK, it is frowned upon:

https://stackoverflow.com/questions/4544234/calling-pthread-cond-signal-without-locking-mutex.

It is best practice to signal the cond variable while holding a lock to the associated mutex.